salt/redash/init.sls

```diff
@@ -1,16 +1,24 @@
+{% from 'lib.sls' import create_pg_database %}
+
 include:
   - docker
 
 # =================================================================================================
 # See https://ocdsdeploy.readthedocs.io/en/latest/deploy/redash.html for installation instructions.
 # =================================================================================================
 
-redash requirements:
-  pkg.installed:
-    - pkgs:
-      - pwgen
-      - postgresql-client-10
-
-/opt/redash:
-  file.directory:
+# Create Redash configuration and directories
+/opt/redash/env:
+  file.managed:
+    - source: salt://redash/files/env
+    - mode: 600
     - makedirs: True
+    - template: jinja
+
+/opt/redash/docker-compose.yml:
+  file.managed:
+    - source: salt://redash/files/docker-compose.yml
+    - mode: 644
+    - template: jinja
+
+{{ create_pg_database( salt['pillar.get']('redash:postgres:database'), salt['pillar.get']('redash:postgres:username') ) }}
```

Comment:
It's unusual to use `pillar.get` here. If the configurations are optional, then we should put an if-statement around the call to the `create_pg_database` macro – otherwise, we're running the macro with null values.

Reply:
Updated to meet pillar.get standards matching docs ✔